crates/bevy_reflect/src/reflect.rs

```diff
@@ -210,6 +210,7 @@ impl Debug for dyn Reflect {
 impl Typed for dyn Reflect {
     fn type_info() -> &'static TypeInfo {
         static CELL: NonGenericTypeInfoCell = NonGenericTypeInfoCell::new();
+        // FIXME : how to build ValueInfo here with TypeName ?
```

Comment:
My guess is we just need to impl `TypeName` for `dyn Reflect` and have it return `dyn Reflect`. This is a special case, but hopefully it won't break anything.